src/libcore/slice/mod.rs

```diff
@@ -2365,6 +2387,209 @@ unsafe impl<'a, T> TrustedRandomAccess for ChunksMut<'a, T> {
     fn may_have_side_effect() -> bool { false }
 }
 
+/// An iterator over a slice in (non-overlapping) chunks (`chunk_size` elements at a
+/// time).
+///
+/// When the slice len is not evenly divided by the chunk size, the last
+/// up to `chunk_size-1` elements will be omitted.
+///
+/// This struct is created by the [`exact_chunks`] method on [slices].
+///
+/// [`exact_chunks`]: ../../std/primitive.slice.html#method.exact_chunks
+/// [slices]: ../../std/primitive.slice.html
+#[derive(Debug)]
+#[unstable(feature = "exact_chunks", issue = "47115")]
+pub struct ExactChunks<'a, T:'a> {
+    v: &'a [T],
+    chunk_size: usize
+}
+
+// FIXME(#26925) Remove in favor of `#[derive(Clone)]`
+#[unstable(feature = "exact_chunks", issue = "47115")]
+impl<'a, T> Clone for ExactChunks<'a, T> {
+    fn clone(&self) -> ExactChunks<'a, T> {
+        ExactChunks {
+            v: self.v,
+            chunk_size: self.chunk_size,
+        }
+    }
+}
+
+#[unstable(feature = "exact_chunks", issue = "47115")]
+impl<'a, T> Iterator for ExactChunks<'a, T> {
+    type Item = &'a [T];
+
+    #[inline]
+    fn next(&mut self) -> Option<&'a [T]> {
+        if self.v.len() < self.chunk_size {
+            None
+        } else {
+            let (fst, snd) = self.v.split_at(self.chunk_size);
+            self.v = snd;
+            Some(fst)
+        }
+    }
+
+    #[inline]
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let n = self.v.len() / self.chunk_size;
+        (n, Some(n))
+    }
+
+    #[inline]
+    fn count(self) -> usize {
+        self.len()
+    }
+
+    #[inline]
+    fn nth(&mut self, n: usize) -> Option<Self::Item> {
+        let (start, overflow) = n.overflowing_mul(self.chunk_size);
+        if start >= self.v.len() || overflow {
```

Comment:
This condition seems wrong, we must test the overflow **before** the test of the `start` is greater or not than the `self.v.len()` because if we have overflowed so the `start` has been wrapped and can be smaller than the `self.v.len()`. And it will be wrong.

We probably want to panic if the computation is impossible and will overflow here, no ?
https://doc.rust-lang.org/std/iter/trait.Iterator.html#panics

Reply:
You still think that an overflow should panic instead of doing nothing? This is currently the same behaviour as for the other `chunks` iterators and what was stabilized for them

Reply:
Yes, I think checked but silent overflows is not a good behavior. But this is a really rare behavior to have an overflow to address the `nth` element, so this is not something we should care of.

Reply:
I agree but I think it's more problematic to have inconsistent behaviour between the different `chunk` iterators (and we can't change the stabilized existing ones). But if there's disagreement I'd be happy to change it

Reply:
If you think that consistency between `chunk` iterators I think we must have consistency between all other iterators and panic if an overflow occurs like the [`Enumerate`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.enumerate) adapter, so it could be a great improvement to update the current implementation of the `Chunks/ChunksMut` to follow this rules. Don't you think ?

Reply:
Can you open an issue about that? I would generally agree (and also think that panicking would be cleaner here) but changing `Chunks/ChunksMut` could be considered a breaking change

Reply:
Here is the issue https://github.com/rust-lang/rust/issues/51254

Reply:
Thanks :)